diff --git a/AllRADecoder/Source/LoudspeakerVisualizer.h b/AllRADecoder/Source/LoudspeakerVisualizer.h
index 1ecb42d43576b1a55bdd1181c07c9868b456dd8d..1f4284ded41f4683306701ca878a6fcf310fd8c4 100644
--- a/AllRADecoder/Source/LoudspeakerVisualizer.h
+++ b/AllRADecoder/Source/LoudspeakerVisualizer.h
@@ -513,7 +513,7 @@ public:
 
     juce::Matrix3D<float> getViewMatrix()
     {
-        juce::Matrix3D<float> translationMatrix (juce::Vector3D<float> (0.0f, 0.0f, -500.0f * zoom)); // move object further away
+        juce::Matrix3D<float> translationMatrix = juce::Matrix3D<float>::fromTranslation(juce::Vector3D<float> (0.0f, 0.0f, -500.0f * zoom)); // move object further away
         juce::Matrix3D<float> tiltMatrix = createRotationMatrix (juce::Vector3D<float> (tilt, 0.0f, 0.0f));
         juce::Matrix3D<float> rotationMatrix = createRotationMatrix (juce::Vector3D<float> (0.0f, yaw, 0.0f));
         return rotationMatrix * tiltMatrix  * translationMatrix;
