diff --git a/app/linux-prebuild.sh b/app/linux-prebuild.sh
index 8d4d4e12c..9c41e4a89 100755
--- a/app/linux-prebuild.sh
+++ b/app/linux-prebuild.sh
@@ -27,9 +27,9 @@ done
 cd "${SCRIPT_DIR}"
 
 if [ ! "$system_libs" == true ]; then
-  "${SCRIPT_DIR}"/linux-pre-vcpkg.sh "${args[@]}"
 	export CC=gcc-12 CXX=g++-12 # Use gcc-12 as the default compiler, since gcc-13 fails to build vcpkg
 	"${SCRIPT_DIR}"/linux-pre-vcpkg.sh "${args[@]}"
+	"${SCRIPT_DIR}"/linux-pre-vcpkg.sh "${args[@]}"
 fi
 
 echo "Compiling native ruby extensions..."
